### DISCUSSION TITLE
"blocks" to direct

### DISCUSSION THREAD

**Discussion by sofia-bg** [UNC-Project-Open-AAC/OS-DPI#195] [Help]
I have this design mostly working but running into a few hitches. The UI starts on what is actually a VSD of the home page with four hotspots that each cover one quadrant. When a quadrant is selected, the UI switches to a grid of only the cells that were in that quadrant of the photo. Users then select individual cells. When a cell is selected with content to display it gets added to the message window and they are returned back to the home VSD. Some of the cells have further actions that take them to yet another VSD with quadrants etc... 

**The Problem:** It is navigating correctly, but the VSD hotspots are not appearing after navigating unless I press the clear button. Any idea what I have done wrong here? 

Also - I have the VSD running through a single tabControl. I tried to do the same with the grids, but found that I was only pulling in the sheet, not the sheet plus only the quadrant. So that is why they are currently in separate tabControls and why there is an empty grid in a tabControl. 

OSDPI File:  [block_to_direct_v1_6.2.23.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/11636058/block_to_direct_v1_6.2.23.zip)

Video example of hotspots failing: 
https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/92817124/d4d1f7a9-3ab3-4ee0-bb66-80e1a4bacb23

Lastly - I know some of the symbols filled wrong in the screenshots I took that are in the VSD levels will fix it later.

**Comment by gbishop** [comment]
Sounds like bugs in my code. I'll look into it.

**Reply by gbishop** [reply]
It is somehow about focus. Clicking anywhere in the designer makes them appear.

**Reply by sofia-bg** [reply]
Hmm - that is not happening for me, it is only when I press clear or clear the state. Even when I first open or load the layout I still need to clear the state. Is it something wrong with my init statement?

**Comment by gbishop** [comment]
I think I've introduced a bug in the VSD with recent changes. Stand by.

**Reply by gbishop** [reply]
I pushed a new version. Refresh your browser and the issue with hotspots should be gone.

I've got to do more testing to see if this version avoids the crashes.

**Reply by sofia-bg** [reply]
works perfectly (famous last words ;) ) thanks!

**Comment by gbishop** [comment]
I didn't follow what you meant by this:

> Also - I have the VSD running through a single tabControl. I tried to do the same with the grids, but found that I was only pulling in the sheet, not the sheet plus only the quadrant. So that is why they are currently in separate tabControls and why there is an empty grid in a tabControl.

I only see a single tab control.

**Reply by gbishop** [reply]
You must mean multiple Tabs. I think you should be able to have only 1 grid and 1 VSD. Each of the grids in the separate Tabs only differ by the value compared to #sheetName. Why not put the value in a $variable? It looks like all you need to do is change the capitalization of either the values in #role or of the sheet names in the spreadsheet to make that work.

**Reply by sofia-bg** [reply]
yes this is what I meant, I tried to do it with one grid and one VSD but got stuck with the grid. I was able to have my actions pull the data from the right sheet but it was showing all 60 rows when I wanted it to only pull from the data marked with the associated quadrant... would love an idea of how to get it to work with only one grid

**Comment by gbishop** [comment]
Your "Actions" page is GREAT! I love how you kept them very simple by using the content.

**Reply by sofia-bg** [reply]
Thanks for noticing! My first approach was starting to get gnarly before I realized that I could do it in a more limited way

**Comment by gbishop** [comment]
![layout](https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/198012/439b16c5-597b-4a86-8ad5-86dd9ce09bb2)
![actions](https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/198012/7bdcbf7a-1e31-45e0-bdff-f8471a8ec04e)

You may ask "Why not send the updated design?". Well, I can't export from
Chrome on my computer right now. A bug in Chrome?

**Reply by gbishop** [reply]
I redid it and exported from Brave.

[block_to_direct_v1_6_2_23-1.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/11638285/block_to_direct_v1_6_2_23-1.zip)

**Reply by sofia-bg** [reply]
Dang I swore that is what I tried last night exactly, but must have been too tired. Thanks! At the very least nice to know I was on the right track.

**Comment by gbishop** [comment]
I sent the design a bit later.

**Reply by sofia-bg** [reply]
yep just wrote at the same time you posted, will check it out
